chore: completely remove the ng scenario runner

Narretz opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Removes the deprecated scenario runner (protractor ancestor)

Closes https://github.com/angular/angular.js/issues/9405

TODO:
- [x] remove / adjust doc info about the scenario runner
- [x] document browserTrigger as part of ngMock (where it has been moved to) 
- [ ] move scenario runner to stand-alone repo (community) before removing it from core? - removing it means no more updates to npm / bower, which could hurt people's build process


**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

Narretz committed:
fix(ngScenario): completely remove the angular scenario runner

The runner has been deprecated and undocumented since 2014:
See commit 8d6d126899d4b1927360599403a7592011243270

Closes #9405

BREAKING CHANGE:

The angular scenario runner end-to-end test framework has been
removed from the project and will no longer be available on npm
or bower starting with 1.7.0.
It was deprecated and removed from the documentation in 2014.
Applications that still use it should migrate to
[Protractor](http://www.protractortest.org).
Technically, it should also be possible to continue using an
older version of the scenario runner, as the underlying APIs have
not changed. However, we do not guarantee future compatibility.

Narretz committed:
docs(*): add browserTrigger docs; remove references to scenario runner

Narretz committed:
refactor($location): remove obsolete workaround for Firefox bug

The bug was fixed in Firefox 48: https://bugzilla.mozilla.org/show_bug.cgi?id=684208,
and only affected the scenario runner

mgol commented on src/ngMock/browserTrigger.js:
```diff
@@ -2,13 +2,23 @@
    * @param {Object=} eventData An optional object which contains additional event data (such as x,y
-   * coordinates, keys, etc...) that are passed into the event when triggered
+   *                            coordinates, keys, etc...) that are passed into the event when
+   *                            triggered.
```
Do changes in this file belong in this PR?

Narretz replied:
It's just a little improvement on the formatting, doesn't hurt, right? 😅 Or do you mean browserTrigger docs in general? We moved bT before in preparation of the scenario runner removal, but never documented it,

gkalpak replied:
Shouldn't we be more specific about what properties are supported in `eventData` what they are used for?

gkalpak commented on src/ng/browser.js:
```diff
@@ -67,7 +67,7 @@ function Browser(window, document, $log, $sniffer) {
   /**
    * @private
-   * Note: this method is used only by scenario runner
+   * Note: this method is used only by protractor / scenario runner
```
Is protractor really using this? I think it uses testability.

Narretz replied:
Testability.whenStable calls this function

gkalpak replied:
I know, but this comment does not belong here. For example, we have discussed in the past that we should make Testability public, which means that anyone could use `whenStable()`. And if we do that, it is very unlikely that someone will remember to update this comment.

If we want to have such a comment, it should be on `whenStable()` imo.

gkalpak commented on src/ng/location.js:
```diff
@@ -938,8 +938,6 @@ function $LocationProvider() {
           if ($location.absUrl() !== $browser.url()) {
             $rootScope.$apply();
-            // hack to work around FF6 bug 684208 when scenario runner clicks on links
-            $window.angular['ff-684208-preventDefault'] = true;
```
This is still used in `browserTrigger`: https://github.com/angular/angular.js/pull/15931/files#diff-6fcc18d824a1325cf54ef306cd2fb7a5R124

Narretz commented:
- removed the comment in $browser
- removed the workaround for Firefox in browserTrigger.
- updated eventData with more info. It's still not perfect, because we use different event creation paths for different browsers / events, but it's better than nothing (since we had no docs before).

gkalpak commented on src/ngMock/browserTrigger.js:
```diff
@@ -9,16 +9,37 @@
+   *  - `y`: y-coordinates for [MouseEvent](https://developer.mozilla.org/docs/Web/API/MouseEvent)
+   *    and [TouchEvent](https://developer.mozilla.org/docs/Web/API/TouchEvent).
+   *
+   *  - `relatedTarget`: the related target for
```
Maybe also link to [MouseEvent.relatedTarget][1].

[1]: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/relatedTarget

gkalpak commented on src/ngMock/browserTrigger.js:
```diff
@@ -9,16 +9,37 @@
+   *
+   *  - `cancelable`: [Event.cancelable](https://developer.mozilla.org/docs/Web/API/Event/cancelable).
+   *    Not applicable to all events.
+
```
Missing `*` :smiley:

gkalpak replied:
Maybe also mention `keyCode`, `charCode` and `which` for `key*` events.

gkalpak reviewed (approved):
I love the docs&nbsp;:heart:
LGTM (with a couple of nits/suggestions) 👍
